servicenow/check.py

```diff
@@ -0,0 +1,208 @@
+# (C) Datadog, Inc. 2010-2016
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+# stdlib
+
+# 3rd party
+import requests
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+from requests.packages.urllib3 import disable_warnings
+
+# Disable https warnings
+disable_warnings(InsecureRequestWarning)
+
+# project
+from checks import AgentCheck, CheckException
+
+EVENT_TYPE = SOURCE_TYPE_NAME = 'servicenow'
+
+
+class ServicenowCheck(AgentCheck):
+
+    INSTANCE_TYPE = "servicenow_cmdb"
+    SERVICE_CHECK_NAME = "servicenow.cmdb.topology_information"
+
+    def check(self, instance):
+        if 'url' not in instance:
+            raise Exception('ServiceNow CMDB topology instance missing "url" value.')
+        if 'basic_auth' not in instance:
+            raise Exception('ServiceNow CMDB topology instance missing "basic_auth" value.')
+
+        basic_auth = instance['basic_auth']
+        if 'user' not in basic_auth:
+            raise Exception('ServiceNow CMDB topology instance missing "basic_auth.user" value.')
+        if 'password' not in basic_auth:
+            raise Exception('ServiceNow CMDB topology instance missing "basic_auth.password" value.')
+
+        relation_types = {}
+
+        basic_auth_user = basic_auth['user']
+        basic_auth_password = basic_auth['password']
+        auth = (basic_auth_user, basic_auth_password)
+
+        base_url = instance['url']
+        batch_size = instance['default_batch_size']
+        instance_key = {"type": self.INSTANCE_TYPE, "url": base_url}
+        instance_tags = instance.get('tags', [])
+
+        default_timeout = self.init_config.get('default_timeout', 5)
+        timeout = float(instance.get('timeout', default_timeout))
+
+        params = {"service_check_done": False, "batch_size": batch_size, "relation_types": relation_types,
```

Comment:
Can you explain to me why all the paramaters end up in this object?
I see the _process_and_cache_relation_types writing into this object, this is an anti-pattern (its a side-effect on a parameter)
If you feel you get too many parameters, I'm much rather see them grouped in a class( which has proper typing), than in a map (which can contain anything)
I think it is best to pair on this because I have a hard time explaining this coding style